### ISSUE TITLE
Prop table does not shows up

### ISSUE THREAD

**Issue by kushalmahajan** [storybookjs/storybook#12212]
Latest storybook fresh install and writing story from any of those mentioned in docs, doesn
't render prop table.

```
"@storybook/addon-actions": "^6.0.16",
    "@storybook/addon-essentials": "^6.0.16",
    "@storybook/addon-links": "^6.0.16",
    "@storybook/react": "^6.0.16",
```

**To Reproduce**

1. `npx sb init`
2. Add Typescript configuration in `.storybook/main` page.
3. Except Button component, no other demo components of Header and Page shows prop tables
4. Restarted the storybook and checked with other methods of writing stories mentioned( Template definition) and it is the same result.

Attaching screenshot and code of my own `Header` component. 
<img width="1493" alt="Screenshot 2020-08-24 at 4 35 55 PM" src="https://user-images.githubusercontent.com/11805936/91038134-e519ae80-e627-11ea-8445-fde5e84c83fa.png">

**Header.tsx**

```
import React from 'react';

interface HeaderProps {
    title: string;
}

export const Header = ({ title }: HeaderProps) => (
  <div>{title}</div>
);

export default Header;

```
**header.stories.tsx**
```
import * as React from 'react';
import Header from './index';

export default {
  title: 'Header/Basic',
  component: Header,
};
export const Primary = () => <Header title="Storybook header" />;

```

**System:**
```
  System:
    OS: macOS 10.15.5
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
  Binaries:
    Node: 12.4.0 - ~/.nvm/versions/node/v12.4.0/bin/node
    Yarn: 1.22.4 - ~/.nvm/versions/node/v12.4.0/bin/yarn
    npm: 6.9.0 - ~/.nvm/versions/node/v12.4.0/bin/npm
  Browsers:
    Chrome: 84.0.4147.135
    Safari: 13.1.1
```

**Comment by kushalmahajan** [comment]
Here is a fun fact. I noticed that Button with same kind of exports is rendering the propTables.However, the columns except Control and Name are empty. It doesn't picks up the comments as written on top of the property inside interface.

So I tried to just add 

```
 argTypes: {
    backgroundColor: { control: 'color' },
  },
``` 
to my header component and it worked.  So I may have given the exact cause here. Seems like the script is not being called for proptable without passing argTypes.
 @shilman
